From c34cf28e84f002c931dc63cdd9cef54850a09952 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 04 Dec 2018 16:57:39 +0200
Subject: [PATCH] [SW-2514] preboot: print the SPL image number

Preboot is the only one during the boot sequence that knows which SPL
image (primary or secondary) will be loaded.
Print this information to console.

Change-Id: Iab5169928a5a29b2e623ea25a65ea4e6f2ca11b4
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot.c b/preboot.c
index d87844e..f3d3b93 100644
--- a/preboot.c
+++ b/preboot.c
@@ -114,6 +114,7 @@
 	unsigned int image_base_offset;
 	unsigned int next_image, next_image_stable, next_image_modifiable;
 	bool try_new_image = false;
+	char *spl_image;
 
 	modifiable_image_pending = readl(mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
 	if (modifiable_image_pending == 3) {
@@ -135,10 +136,17 @@
 			next_image = 0;
 	}
 
-	if (!next_image) /* Primary */
+	if (!next_image) { /* Primary */
 		image_base_offset = 0;
-	else /* Secondary */
+		spl_image = "primary";
+	} else {/* Secondary */
 		image_base_offset = 0x800000;
+		spl_image = "secondary";
+	}
+
+	hl_preboot_puts_raw("Preparing to run SPL ");
+	hl_preboot_puts_raw(spl_image);
+	hl_preboot_puts("\n");
 
 	*spl_off = image_base_offset + spl_offset;
 	writel((image_base_offset + uboot_offset), mmPSOC_GLOBAL_CONF_SCRATCHPAD_30);
